Dockerfile

```diff
@@ -1,75 +1,64 @@
-ARG GO_VERSION=1.17.5
-FROM golang:$GO_VERSION-bullseye as builder
+FROM ubuntu:18.04 as builder
 
-ARG CHANNEL=nightly
-ARG URL=
-ARG BRANCH=
-ARG SHA=
+ARG GO_VERSION="1.17.13"
+
+ARG CHANNEL
+ARG URL
+ARG BRANCH
+ARG SHA
+ARG TARGETARCH
+
+ADD https://go.dev/dl/go${GO_VERSION}.linux-${TARGETARCH}.tar.gz /go.tar.gz
 
 # Basic dependencies.
-ENV HOME /node
-ENV DEBIAN_FRONTEND noninteractive
+ENV HOME="/node" DEBIAN_FRONTEND="noninteractive" GOPATH="/node"
+
 RUN apt-get update && \
-    apt-get install -y \
+    apt-get install -y --no-install-recommends \
+    ca-certificates \
     apt-utils \
     bsdmainutils \
     curl \
     git \
-    git-core \
-    python3
+    && rm -rf /var/lib/apt/lists/* && \
+    \
+    tar -C /usr/local -xzf /go.tar.gz && \
+    rm -rf /go.tar.gz
+
+ENV PATH="/usr/local/go/bin:${PATH}"
 
 COPY ./docker/files/ /node/files
 COPY ./installer/genesis /node/files/run/genesis
 COPY ./cmd/updater/update.sh /node/files/build/update.sh
-COPY ./installer/config.json.example /node/files/build/config.json
-
-RUN find /node/files
+COPY ./installer/config.json.example /node/files/run/config.json.example
 
 # Install algod binaries.
 RUN /node/files/build/install.sh \
-    -p "/node/bin" \
+    -p "${GOPATH}/bin" \
     -d "/node/data" \
     -c "${CHANNEL}" \
     -u "${URL}" \
     -b "${BRANCH}" \
     -s "${SHA}"
 
-# Copy binaries into a clean image
-# TODO: We don't need most of the binaries.
-#       Should we delete everything except goal/algod/algocfg/tealdbg?
 FROM debian:bullseye-slim as final
-COPY --from=builder "/node/bin/" "/node/bin"
-COPY --from=builder "/node/data/" "/node/dataTemplate"
-COPY --from=builder "/node/files/run" "/node/run"
 
-ENV BIN_DIR="/node/bin"
-ENV PATH="$BIN_DIR:${PATH}"
-ENV ALGOD_PORT=8080
-ENV ALGORAND_DATA="/algod/data"
-RUN mkdir -p "$ALGORAND_DATA"
-WORKDIR /node/data
```

Comment:
WORKDIR should be updated to "/algod" or "/algod/data" so that when you enter the container it puts you in the right spot.